src/movepick.cpp

```diff
@@ -63,9 +62,8 @@ MovePicker::MovePicker(const Position& p, Move ttm, Depth d, const ButterflyHist
 
   assert(d > 0);
 
-  stage = pos.checkers() ? EVASION_TT : MAIN_TT;
-  ttMove = ttm && pos.pseudo_legal(ttm) ? ttm : MOVE_NONE;
-  stage += (ttMove == MOVE_NONE);
+  stage = pos.checkers() ? EVASION_INIT : CAPTURE_INIT;
+  useTTM = (ttMove = ttm && pos.pseudo_legal(ttm) ? ttm : MOVE_NONE);
```

Comment:
Do we still need the ternary operator here?

Reply:
Excellent observation.  Yes.  I think that can go away now as well.  Just a sec.

Reply:
Darn. . doesn't look like it.  If the given ttm is not legal, ttMove has to be set to MOVE_NONE or the bench changes.

Reply:
also, since we now unconditionally assign to ttMove, it can be done in the initializer list in all three constructors, which further simplifies the conditional.

Reply:
OK, so my last comment is not quite right.

Reply:
actually, I'm not sure why. ttm shouldn't be used if it is not pseudo-legal? It might result in an other generated move (that is not pseudo legal either, being skipped later in the movepicker)?

Reply:
You may be right.  Should I further simplify this way and test it?

Reply:
I also considered ttMove = ttm * (conditional checks).  If false, ttMove would be MOVE_NONE.  If true, ttm.  Perhaps that is too cryptic.

Reply:
That latter suggestion is too cryptic. Before testing on fishtest, you can maybe try to understand if my suggested reason is indeed correct. Afterwards, as this becomes a functional change, it should be retested.

Reply:
OK.  Will do.

Reply:
In master, if ttm fails to meet the conditions (e.g., pseudo_legal), ttMove is set to MOVE_NONE.  In this case, we may still return the ttMove (even if not pseudo_legal) because we no longer track what it was.

Thus, it seems BETTER to always assign ttm to ttMove.  This way, we NEVER return the ttMove regardless of whether it met the prescribed conditions (i.e., pseudo_legal, etc.).

If that seems right to you, should I test that on the framework?

Reply:
I think that's correct indeed. Note that in the probcut constructor stage could also be set in the initializer list.

Reply:
https://tests.stockfishchess.org/tests/view/5e821a76e42a5c3b3ca2edfa

Reply:
Test failed, so I'm not sure we can remove this ternary operator. . yet.  I'll keep thinking about it and make a different PR if I think of something.